license.go

```diff
@@ -0,0 +1,92 @@
+package pagerduty
+
+import (
+	"context"
+
+	"github.com/google/go-querystring/query"
+)
+
+type License struct {
+	APIObject
+	Name                 string   `json:"name"`
+	Description          string   `json:"description"`
+	ValidRoles           []string `json:"valid_roles"`
+	RoleGroup            string   `json:"role_group"`
+	Summary              string   `json:"summary"`
+	CurrentValue         int      `json:"current_value"`
+	AllocationsAvailable int      `json:"allocations_available"`
+}
+
+type LicenseAllocated struct {
+	APIObject
+	Name        string   `json:"name"`
+	Description string   `json:"description"`
+	ValidRoles  []string `json:"valid_roles"`
+	RoleGroup   string   `json:"role_group"`
+	Summary     string   `json:"summary"`
+}
+
+type LicenseAllocation struct {
+	AllocatedAt string `json:"allocated_at"`
+	User        APIObject
+	License     LicenseAllocated `json:"license"`
+}
+
+type ListLicensesResponse struct {
+	Licenses []License `json:"licenses"`
+}
+
+type ListLicenseAllocationsResponse struct {
+	APIListObject
+	LicenseAllocations []LicenseAllocation `json:"license_allocations"`
+}
+
+type ListLicenseAllocationsOptions struct {
+	Limit  int `url:"limit,omitempty"`
+	Offset int `url:"offset,omitempty"`
+}
+
+func (c *Client) ListLicenses() (*ListLicensesResponse, error) {
+	return c.ListLicensesWithContext()
+}
+
+func (c *Client) ListLicenseAllocations(o ListLicenseAllocationsOptions) (*ListLicenseAllocationsResponse, error) {
+	return c.ListLicenseAllocationsWithContext(o)
+}
+
+func (c *Client) ListLicensesWithContext() (*ListLicensesResponse, error) {
```

Comment:
The *WithContext() methods should accept a `context.Context` parameter. The newly added Event Orchestration APIs should give you an example of this: https://github.com/PagerDuty/go-pagerduty/blob/main/event_orchestration.go#L73-L90.

Reply:
Could we also remove the non-context methods since that is now a deprecated pattern?